packages/utils/src/ratelimit.ts

```diff
@@ -85,7 +85,16 @@ export function updateRateLimits(
         updatedRateLimits.all = now + delay;
       } else {
         for (const category of categories.split(';')) {
-          updatedRateLimits[category] = now + delay;
+          if (category === 'metric_bucket') {
+            const namespaces = limit.split(':', 5)[4] ? limit.split(':', 5)[4].split(';') : null;
```

Comment:
can we combine this `limit.split` call with the one above in `const [retryAfter, categories]`?

also generally we prefer against using `null`, `undefined` is returned by default when array access returns nothing, and it helps reduce bundle size generally.